support_for_advertising_use_cases.md

```diff
@@ -529,51 +531,55 @@ This use case is common in sales period. Note that sales period can have very pr
 
 Advertisers want to optimize the look and feel of their ads to the audience and to the environment. This helps to drive more performance and improve their overall brand recognition.
 
-This can be done programmatically using historical aggregated and/or personal user data, and contextual information. 
+This can be done programmatically using historical aggregated and/or personal user data, and contextual information.
 
 ## Coupon Management
 
 As an Advertiser I want to be able to add coupons, or any form of targeted discounts to my creatives on specific period of times, potentially for specific products.
 
+# ML Applications
 
-## Recommendation
+Machine Learning Models are very important part of optimal investment in advertising. They are used by many participants in three major areas:
 
-Large e-Commerce websites may have product catalogs with millions of items. They have an interesting use-case. When running ads to promote items from their vast catalogs, which product should they show?
-Today, much of this is mainly powered by data about which items a person has previously browsed.
+## Item recommendations
 
-Typical sources of recommendation would be :
-- Historical Products (compared to historical products browsed): abandoned cart, etc.
-- Similar products (compared to historical products browsed)
-- Complementary Products (compared to historical products browsed)
-- ...
+When there are many items to choose from (products/content/messages/apps) before rendering ad, which one should be shown to which person and in which ad? This produces more useful ads for consumers. Users are less annoyed with ads because they will more likely see items they would like to buy/convert or see less products they already bought/installed or they are not interested to buy/install.
```

Comment:
I actually prefer the old wording, and would rather not change this.

What I like about the old wording is that it's more concrete. It helps the reader picture the use-case and understand. Your new wording is very abstract and vague. I worry it doesn't necessarily help the reader understand.

Honestly, I'm not really sure what you mean when you talk about selecting from many messages. What is this use-case? Is it worth calling out as a separate use-case?

Reply:
> I actually prefer the old wording, and would rather not change this. What I like about the old wording is that it's more concrete. It helps the reader picture the use-case and understand. Your new wording is very abstract and vague. I worry it doesn't necessarily help the reader understand.

Valid point and it looks to me that  @appascoe also has a similar point of view.

My main objective here was to point out that item recommendations are a greater concept than just a product which might be categorized underneath and that they have multidimensional application.
E-commerce indeed is a concrete example with such items as products (shoes,bikes etc). And the previous meaning was perfectly capturing it.
Other great examples can be:

- Publishers are willing to show recommendations based on historical content consumption.
- Brands willing to show copy recommendations according to the user's stage of the marketing funnel, based on historical data.
- App Publishers are willing to promote install other apps from their catalog, based on historical data.
- …

@benjaminsavage  you have any proposals on how we can make it more general and applicable for different use cases?
> Honestly, I'm not really sure what you mean when you talk about selecting from many messages. What is this use-case? Is it worth calling out as a separate use-case?

I placed an example in the above explanation
Probably changing  “message” to “copy” will be more concrete and straightforward.
 
Maybe a great idea will be to try to catalog specialized types of advertisers in separate heading and call out the most characteristic and important from the perspective of the ad economy?

It might be useful for everyone to explain the specific needs of eCommerce, Brands, Apps Publishers etc. They are separate use cases from a business perspective. Also might be a great activation point to hear advertisers point of view? @benjaminsavage WDYT?

Reply:
I see what you're saying. I think the difference is that I would conceptualize of these things as either "Ad Selection" or "Format Automation". 

I usually think of the mobile app install ad use-case is a matter of selecting which ad. Perhaps we can call that out specifically as an example in the "Ad Selection" use-case?

As for the changing ad-copy based on the stage in the marketing funnel, that I would categorize as "Format Automation". The same ad is selected, but the advertiser has multiple variants of how to display it. Perhaps we can add that as an example in the "Format Automation" section?

As for the publishers showing recommendations based on historical content consumption, I'm less clear on this use-case. By "suggestions" do you mean "ads"? Are you referring to the use of historical first-party interactions with that publisher as a feature in ad selection?

Reply:
I am not really sure what criteria are you using to differentiate those key uses in relation to concept  (and I would be happy if you can share it here to have clear understanding on both ends)
In my understanding all above mentioned examples might be achieved by item recommendation in understanding that item is a component of an ad. 
And if we think about an ad as a top rendered container in a client browser built from smaller components (items) like visualized products, copy, content, apps etc. 
Additionally my understanding is that we might have different types of ads like display banners, native or video. Example: You need to use different components for banner ad and for native ad.
Also formatting would be defining variants how an ad might be rendered from style perspective to make it more attractive for users. Example some users are more likely clicking on the left side, others on the right side of the banner or some style in banner converts more likely than others. Following example might be when users are more likely clicking on animated ads than static.
So your proposals in my understanding are mixing up for me containers with components.
 
Nevertheless, I believe it is important and highly valuable for all group members  to keep above mentioned use cases in document with comment allowing to understand why they are assigned to particular concepts. 
@benjaminsavage could you please advice?
 
> As for the publishers showing recommendations based on historical content consumption, I'm less clear on this use-case. By "suggestions" do you mean "ads"? Are you referring to the use of historical first-party interactions with that publisher as a feature in ad selection?
 
This use case represents when a publisher acts like an advertiser and they would like to use historical first-party interactions to build an ad with recommendation to their content. Example huge video publishers would like to target interest groups of their web page to revisit and see some new great video content items they have which are relevant and useful for those visitors.

Reply:
I understand where you're coming from. It sounds like in your ads system, you think of "ad" as the thing that is rendered on the page, and have generalized all the aspects of that ad as "items" and use ML to select the "items" that are combined to form the "ad". 

That makes sense, but it's pretty different from how our ads system works - which is probably why I find this terminology confusing =). I guess my main concern is that "Item recommendations" sounds very vague. 

Here's a recommendation:

1.) Ad-campaign selection.
e.g. There are many advertising campaigns to choose from. These might be promoting different types of conversion events (e.g. some are promoting mobile app installs, some are promoting purchases on websites, some are just optimized to generate link-clicks, some are optimized to generate video-views.)
2.) Ad-component selection.
Ads are composed of multiple components (e.g. the title, description, image, video, etc.). Some ads are promoting particular products. Once an ad-campaign is selected, ML is used to select which components to use in each particular ad impression. (e.g. which products to highlights, which title text to use, which image to use). These options are constrained by what the advertiser has provided as options for their ad-campaign.
3.) Formatting customization.
Once the ad-campaign and components are selected, choices remain about how exactly to render that set of components (e.g. button on the right or the left).

@lukwlodarczyk - what do you think of this set of categories?

Reply:
Looks comprehensive to me. And will be happy to do the final commit in this pull request. I will re-request a review soon.
Happy to see the great outcome of this thread!